crates/pink/src/runtime/extension.rs

```diff
@@ -79,13 +80,14 @@ pub fn get_side_effects() -> ExecSideEffects {
 }
 
 /// Contract extension for `pink contracts`
+#[derive(Default)]
 pub struct PinkExtension;
 
 impl ChainExtension<super::PinkRuntime> for PinkExtension {
-    fn call<E: Ext>(func_id: u32, env: Environment<E, InitState>) -> Result<RetVal, DispatchError>
+    fn call<E: Ext>(&mut self, env: Environment<E, InitState>) -> ExtResult<RetVal>
     where
         <E::T as SysConfig>::AccountId:
-            UncheckedFrom<<E::T as SysConfig>::Hash> + AsRef<[u8]> + Clone,
+        UncheckedFrom<<E::T as SysConfig>::Hash> + AsRef<[u8]> + Clone,
     {
```

Comment:
Now, we need to ensure `env.ext_id() == 0`

Reply:
add a guard
```
        if env.ext_id() != 0 {
            error!(target: "pink", "Unknown extension id: {:}", env.ext_id());
            return Err(DispatchError::Other(
                "PinkExtension::call: unknown extension id",
            ))
        }
```